client/my-sites/stats/activity-log-confirm-dialog/index.jsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import Button from 'components/button';
+import Gridicon from 'gridicons';
+
+class ActivityLogConfirmDialog extends Component {
+	static propTypes = {
+		siteName: PropTypes.string.isRequired,
+	};
+
+	static defaultProps = {
+		siteName: 'YourAwesomeSite', // FIXME: real site name
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			showDialog: true,
+		};
+
+		this.onCloseDialog = this.onCloseDialog.bind( this );
+	}
+
+	renderButtons() {
+		const { translate } = this.props;
+		return (
+			<div>
+				<Button onClick={ this.onCloseDialog }>
+					{ translate( 'Cancel' ) }
+				</Button>
+				<Button primary scary onClick={ this.onCloseDialog }>
+					{ translate( 'Restore' ) }
+				</Button>
+			</div>
+		);
+	}
+
+	onCloseDialog() {
+		this.setState( { showDialog: false } );
+	}
```

Comment:
To avoid binding in `constructor`, just use:

```js
onCloseDialog = () => this.setState( { showDialog: false } );
```

Nit: I think `handle` rather than `on` prefix avoids confusion.